5.1.thats_the_way.py

```diff
@@ -0,0 +1,20 @@
+import os
+
+
+def get_files_from_dir(directory):
+    """
+    Returns a list of all the files in a given directory that start with 'deep'
+    :param directory: The directory for searching
+    :return: List of the files in the directory that start with 'deep'
+    """
+    return [f for f in os.scandir(directory) if os.path.isfile(f) and f.startswith('deep')]
```

Comment:
Add error handling for invalid directories: The os.scandir function may raise a FileNotFoundError if the given directory does not exist. Consider adding error handling to catch this exception and provide meaningful feedback to users.